vlm/src/main/scala/geotrellis/contrib/vlm/geotiff/GeoTiffReprojectRasterSource.scala

```diff
@@ -80,27 +79,62 @@ case class GeoTiffReprojectRasterSource(
   }
 
   override def readExtents(extents: Traversable[Extent], bands: Seq[Int]): Iterator[Raster[MultibandTile]] = {
-    val bounds = extents.map(rasterExtent.gridBoundsFor(_, clamp = true))
-    readBounds(bounds, bands)
+    val geoTiffTile = closestTiffOverview.tile.asInstanceOf[GeoTiffMultibandTile]
```

Comment:
Hm, probably some changes have to be introduced for the GDAL backend as well :/ I could fix a GeoTiff backend (but it also sounds like a workaround). GDAL produces weirdly sized windows (tile sizes are larger / not according to the expected tile layout)